samplers/parser.go

```diff
@@ -122,29 +122,57 @@ func ConvertMetrics(m *ssf.SSFSpan) ([]UDPMetric, error) {
 // span and returns metrics that can be determined from that
 // span. Currently, it converts the span to a timer metric for the
 // duration of the span.
-func ConvertIndicatorMetrics(span *ssf.SSFSpan, timerName string) (metrics []UDPMetric, err error) {
-	if !span.Indicator || timerName == "" {
+func ConvertIndicatorMetrics(span *ssf.SSFSpan, indicatorTimerName, objectiveTimerName string) (metrics []UDPMetric, err error) {
+	if !span.Indicator {
 		// No-op if this isn't an indicator span
 		return
 	}
 
 	end := time.Unix(span.EndTimestamp/1e9, span.EndTimestamp%1e9)
 	start := time.Unix(span.StartTimestamp/1e9, span.StartTimestamp%1e9)
-	tags := map[string]string{
-		"service": span.Service,
-		"error":   "false",
-	}
-	if span.Error {
-		tags["error"] = "true"
+	duration := end.Sub(start)
+
+	if indicatorTimerName != "" {
+		tags := map[string]string{
+			"service": span.Service,
+			"error":   "false",
+		}
+		if span.Error {
+			tags["error"] = "true"
+		}
+		ssfTimer := ssf.Timing(indicatorTimerName, duration, time.Nanosecond, tags)
+		ssfTimer.Name = indicatorTimerName // Ensure the name is free from any name prefixes, like "veneur."
+
+		timer, err := ParseMetricSSF(ssfTimer)
+		if err != nil {
+			return metrics, err
+		}
+		metrics = append(metrics, timer)
 	}
-	ssfTimer := ssf.Timing(timerName, end.Sub(start), time.Nanosecond, tags)
-	ssfTimer.Name = timerName // Ensure the name is free from any name prefixes, like "veneur."
 
-	timer, err := ParseMetricSSF(ssfTimer)
-	if err != nil {
-		return metrics, err
+	if objectiveTimerName != "" {
+		tags := map[string]string{
+			"service":          span.Service,
+			"objective":        span.Name,
```

Comment:
do we care about spans with empty name?

Reply:
It isn't something we insist exists for `ValidTrace`. That said, I can't imagine why that would ever be desireable and *not* a bug. Should we just add that to `ValidTrace` and insist that a span must have a non-empty name?

(Also, is `ValidTrace` even called before this point somewhere in the call chain? If not, it should be).

Reply:
hmm, it doesn't make sense to run ValidTrace in the metrics extraction sink (since there might be ssf packets with no span), but it probably _does_ make sense to run it at this specific place (since indicator metrics only make sense for ssf packets that do have a span). will fix